### PR TITLE
[4.5] refactor: fix types

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- fix types for #8072

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/MigrationRunner.php [hunk 1]
```diff
@@ -752,7 +752,7 @@ public function getBatchEnd(int $batch): string
             ->get()
             ->getResultObject();
 
-        return count($migration) ? $migration[0]->version : 0;
+        return count($migration) ? $migration[0]->version : '0';
```
Comment (MGatner) [inline system/Database/MigrationRunner.php:755]: Might as well. 
```suggestion
        return $migration === [] ? '0' : $migration[0]->version;
```

File: system/Test/CIUnitTestCase.php [hunk 2]
```diff
@@ -439,13 +439,13 @@ public function assertHeaderNotEmitted(string $header, bool $ignoreCase = false)
...
      *
-     * @param mixed $actual
+     * @param float|int $actual
      *
      * @throws Exception
      */
     public function assertCloseEnough(int $expected, $actual, string $message = '', int $tolerance = 1)
     {
-        $difference = abs($expected - (int) floor($actual));
+        $difference = abs($expected - (int) floor((float) $actual));
```
Comment (MGatner) [inline system/Test/CIUnitTestCase.php:448]: This seems like it should be either/or. `floor()` supports `int` input so why the cast? unless the docblock type isn't certain?

**Review by MGatner — CHANGES_REQUESTED** [review]

File: system/Database/MigrationRunner.php (hunk 1 above)
Comment (kenjis) [reply system/Database/MigrationRunner.php]: Good!

File: system/Test/CIUnitTestCase.php (hunk 2 above)
Comment (kenjis) [reply system/Test/CIUnitTestCase.php]: Yes, I removed `(float)`.

**Review by MGatner — APPROVED** [review]

**Pull request merged**